pandas-stubs/core/series.pyi

```diff
@@ -216,13 +217,43 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
     @overload
     def __new__(
         cls,
-        data: IntervalIndex[IntervalT],
+        data: IntervalIndex[Interval[int]],
         index: Axes | None = ...,
         dtype=...,
         name: Hashable | None = ...,
         copy: bool = ...,
         fastpath: bool = ...,
-    ) -> Series[IntervalT]: ...
+    ) -> Series[Interval[int]]: ...
+    @overload
```

Comment:
maybe also reference the mypy issue again

Reply:
> maybe also reference the mypy issue again

Looks like I included changes from the other PR in this one, so I have removed those changes related to `Series[IntervalT]`

@twoertwein  If OK with you, can you approve and merge?